File: 25 Coding Challenge I/JavaScript/Challenge14.js
// Challenge 14: Print the first 10 Fibonacci numbers without using recursion.

let f0 = 0;
let f1 = 1;

console.log(f0);
console.log(f1);

for( let i = 2; i < 10; i++){
   let fi = f1 + f0;
   console.log(f1);
   
   // swaping
   f0 = f1;
   f1 = fi;

}

File: 50 Coding Challenge Part I/Java/src/com/mofirojean/Challenge14.java
package com.mofirojean;

public class Challenge14 {

    public static void main(String[] args){
        fibonacci();
    }

    public static void fibonacci() {
        int firstNumber = 0; // define the initial values of any fibonacci series
        int secondNumber = 1;
        int count = 1;
        System.out.println(firstNumber);
        System.out.println(secondNumber);
        while (count < 8) {
            int number = firstNumber + secondNumber; //the new number in the series is the sum of the previous two numbers
            firstNumber = secondNumber; // the new values of first and second numbers are set
            secondNumber = number;
            System.out.println(number);
            count++;
        }
    }
}
